nixos/modules/virtualisation/libvirtd.nix

```diff
@@ -177,17 +185,23 @@ in {
           ln -s --force "$emulator" /run/libvirt/nix-emulators/
         done
 
+        mkdir -p /run/libvirt/nix-helpers
+        for helper in libexec/qemu-bridge-helper bin/qemu-pr-helper; do
+          ln -s --force ${cfg.qemuPackage}/$helper /run/libvirt/nix-helpers/
+        done
+
         ${optionalString cfg.qemuOvmf ''
-            mkdir -p /run/libvirt/nix-ovmf
-            ln -s --force ${pkgs.OVMF.fd}/FV/OVMF_CODE.fd /run/libvirt/nix-ovmf/
-            ln -s --force ${pkgs.OVMF.fd}/FV/OVMF_VARS.fd /run/libvirt/nix-ovmf/
+          ln -s --force ${pkgs.OVMF.fd}/FV/OVMF_CODE.fd /run/libvirt/nix-ovmf/
+          ln -s --force ${pkgs.OVMF.fd}/FV/OVMF_VARS.fd /run/libvirt/nix-ovmf/
         ''}
       '';
 
       serviceConfig = {
         Type = "notify";
         KillMode = "process"; # when stopping, leave the VMs alone
         Restart = "no";
+        LogsDirectory = "libvirt/qemu";
+        RuntimeDirectory = [ "libvirt/dnsmasq" "libvirt/nix-ovmf" ];
```

Comment:
does it make directories in /run or /var/lib ?
(i am not sure how does dnsmasq work and which directories it needs, but it seems that there was ```/var/lib/libvirt/dnsmasq``` and ```/run/libvirt/nix-ovmf``` before and now "libvirt/dnsmasq" "libvirt/nix-ovmf" are closer to each other)

Reply:
You're right - nicely spotted.